### PR TITLE
docs: user general pouchlinter to linter pouch's code

### PR DESCRIPTION
**Pull request by allencloud**
Signed-off-by: Allen Sun <allensun.shl@alibaba-inc.com>

### Ⅰ. Describe what this PR did

user general pouchlinter to linter pouch's code.

For more info about pouchlinter, please check https://github.com/pouchcontainer/pouchlinter

The doc change in test.md and pouch_with_plugin.md is according to markdowinlint rule https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md046---code-block-style .

### Ⅱ. Does this pull request fix one issue?

none

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

no need

### Ⅳ. Describe how to verify it
none

### Ⅴ. Special notes for reviews
none

### CODE REVIEW

**Comment by allencloud** [comment]
The gometalinter works. @fuweid 
Do we need to remove the shell scripts in folder ./hack/...... ?
I am not sure if they are still useful?

**Comment by fuweid** [comment]
Some scripts are used to help developer to install dependences easier. I think we can keep it.

File: .circleci/config.yml [hunk 1]
```diff
@@ -25,16 +26,19 @@ jobs:
...
+          name: validate swagger.yml
+          command: |
+            swagger validate "/go/src/github.com/alibaba/pouch/apis/swagger.yml"
+            
       - run: 
-          name: use check target in Makefile to check code
-          command: make check
+          name: validate go code with gometalinter
+          command: |
+            gometalinter --disable-all --skip vendor -E gofmt -E goimports -E golint -E ineffassign -E misspell -E vet -E megacheck ./...
```
Comment (fuweid) [inline .circleci/config.yml]: one more question, could we just use `make check` here, since the image provides the binary?

Comment (allencloud) [reply .circleci/config.yml]: I think both are OK, while if we use `make check`, the calling path is a little bit long. For the test part and ci part, in my opinion, **simplicity** is the most important thing.

**Pull request merged**